remoteappmanager/handlers/home_handler.py

```diff
@@ -112,7 +112,9 @@ def _actionhandler_start(self, options):
 
         # The server is up and running. Now contact the proxy and add
         # the container url to it.
-        urlpath = yield self.application.reverse_proxy.add_container(container)
+        urlpath = self.application.urlpath_for_object(container)
```

Comment:
Because there should only be one authoritative source for a given information. And with containers the container itself can't know. It can only know its own relative path, but to know the absolute one, the application is authoritative, so it should be inquired, and it should be inquired with an interface that is generic.

I am not saying this method is the perfect solution. I am saying is that we would be scattering around logic that is not the concern of the handler.